src/main/webapp/dataset.xhtml

```diff
@@ -777,7 +777,7 @@
                     <!-- Create/Edit editMode -->
                     <ui:fragment rendered="#{DatasetPage.editMode == 'METADATA' or DatasetPage.editMode == 'CREATE'}">
                         <p:focus context="datasetForm"/>
-                        <div class="form-group">
+                        <div class="form-group" jsf:rendered="#{DatasetPage.hasDataversesToChoose and DatasetPage.editMode == 'CREATE'}">
```

Comment:
FWIW: Some additional small performance tweaks that could be made:

- I think it's true that JSF will evaluate left to right, so if DatasetPage.editMode == 'CREATE' were first and was false, the hasDataverseToChoose query would not get made. 
- JSF can call methods multiple times (up to 7!) so caching the boolean answer to hasMultipleDataversesToChoose in the DatasetPage would also improve performance (probably more important if/when there's also a permission check to see if the user has permissions to add datasets in more than one dataverse.).

Reply:
778 limits it to showing only when the mode is CREATE (or METADATA) so I guess changing 780 would only help in the METADATA case. Definitely minor.